lib/utils/visUtil.cpp

```diff
@@ -364,6 +364,13 @@ double StatTracker::get_std_dev() {
     return std_dev;
 }
 
+double StatTracker::get_current() {
+    std::lock_guard<std::mutex> lock(tracker_lock);
```

Comment:
That's a handy class, you don't even have to unlock the mutex explicitly like in C.

Reply:
The mutex has been changed to recursive_mutex.
```suggestion
    std::lock_guard<std::recursive_mutex> lock(tracker_lock);
```